treeherder/etl/seta.py

```diff
@@ -78,3 +85,56 @@ def job_priorities_to_jobtypes():
         jobtypes.append(jp.unique_identifier())
 
     return jobtypes
+
+
+def get_reference_data_names(project, build_system):
+    '''
+    We want all reference data names for every task that runs on a specific project.
+
+    For example: "test-linux64/opt-mochitest-webgl-e10s-1"
+    '''
+    # we cache the reference data names in order to reduce API calls
+    cache_key = '{}-{}-ref_data_names_cache'.format(project, build_system)
+    ref_data_names_map = cache.get(cache_key)
```

Comment:
We move the caching logic within this module.

See original location outside of the function in here:
https://github.com/mozilla/treeherder/pull/6002/files#diff-585361d31307bed77c0dc7fd8a411fd2L42